rust/query/mod.rs

```diff
@@ -25,84 +25,41 @@ use std::fmt;
 use crate::pattern::*;
 use crate::{enum_getter, var, ErrorMessage};
 
+mod typeql_delete;
+pub use typeql_delete::*;
+
+mod typeql_insert;
+pub use typeql_insert::*;
+
 mod typeql_match;
 pub use typeql_match::*;
 
+mod typeql_update;
+pub use typeql_update::*;
+
 #[derive(Debug, Eq, PartialEq)]
 pub enum Query {
     Match(TypeQLMatch),
-    Dud(()),
+    Insert(TypeQLInsert),
+    Delete(TypeQLDelete),
+    Update(TypeQLUpdate),
 }
 
 impl Query {
     enum_getter!(into_match, Match, TypeQLMatch);
-}
-
-impl MatchQueryBuilder for Query {
-    fn get<T: Into<String>, const N: usize>(self, vars: [T; N]) -> Self {
-        use Query::*;
-        match self {
-            Match(query) => Match(query.get(vars)),
-            _ => todo!(),
-        }
-    }
-
-    fn sort(self, sorting: impl Into<Sorting>) -> Self {
-        use Query::*;
-        match self {
-            Match(query) => Match(query.sort(sorting)),
-            _ => todo!(),
-        }
-    }
-
-    fn limit(self, limit: usize) -> Self {
-        use Query::*;
-        match self {
-            Match(query) => Match(query.limit(limit)),
-            _ => todo!(),
-        }
-    }
-
-    fn offset(self, offset: usize) -> Self {
-        use Query::*;
-        match self {
-            Match(query) => Match(query.offset(offset)),
-            _ => todo!(),
-        }
-    }
+    enum_getter!(into_insert, Insert, TypeQLInsert);
+    enum_getter!(into_delete, Delete, TypeQLDelete);
+    enum_getter!(into_update, Update, TypeQLUpdate);
 }
 
 impl fmt::Display for Query {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         use Query::*;
         match self {
             Match(query) => write!(f, "{}", query),
-            _ => todo!(),
+            Insert(query) => write!(f, "{}", query),
+            Delete(query) => write!(f, "{}", query),
+            Update(query) => write!(f, "{}", query),
         }
     }
 }
-
-pub trait MatchQueryBuilder {
-    fn get<T: Into<String>, const N: usize>(self, vars: [T; N]) -> Self;
-    fn sort(self, sorting: impl Into<Sorting>) -> Self;
-    fn limit(self, limit: usize) -> Self;
-    fn offset(self, offset: usize) -> Self;
-}
-
-impl<U: MatchQueryBuilder> MatchQueryBuilder for Result<U, ErrorMessage> {
-    fn get<T: Into<String>, const N: usize>(self, vars: [T; N]) -> Self {
-        Ok(self?.get(vars))
-    }
-
-    fn sort(self, sorting: impl Into<Sorting>) -> Self {
-        Ok(self?.sort(sorting))
-    }
-
-    fn limit(self, limit: usize) -> Self {
-        Ok(self?.limit(limit))
-    }
-
-    fn offset(self, offset: usize) -> Self {
-        Ok(self?.offset(offset))
-    }
-}
```

Comment:
This trait no longer needs to be implemented for `Query` now that builders return the most specific type, so this definition is moved to query/typeql_match instead